perf/security/sds-tests/ingress-sds/setup_test.sh

```diff
@@ -0,0 +1,153 @@
+    ingress_host=$(kubectl -n istio-system get service istio-ingressgateway -o jsonpath='{.status.loadBalancer.ingress[0].ip}')
+    echo "secure ingress port: ${secure_ingress_port}"
+    echo "ingress host: ${ingress_host}"
+}
+
+function prepare_ingress_secret() {
+    # shellcheck disable=SC2086
+    openssl req -x509 -sha256 -nodes -days 365 -newkey rsa:2048 -subj '/O=example Inc./CN=example.com' -keyout ${wd}/example.com.key -out ${wd}/example.com.crt
+    # shellcheck disable=SC2086
+    kubectl create -n istio-system secret generic ingress-root --from-file=key=${wd}/example.com.key --from-file=cert=${wd}/example.com.crt --cluster "${CLUSTER}"
+    # shellcheck disable=SC2004
+    for ((id=1; id<=${NUM}; id++)); do
+        credential_name="httpbin-credential-${id}"
+        # shellcheck disable=SC2086
+        openssl req -out ${wd}/httpbin.example.com.csr -newkey rsa:2048 -nodes -keyout ${wd}/httpbin.example.com.key -subj "/CN=httpbin-${id}.example.com/O=httpbin organization"
+        # shellcheck disable=SC2086
+        openssl x509 -req -days 365 -CA ${wd}/example.com.crt -CAkey ${wd}/example.com.key -set_serial 0 -in ${wd}/httpbin.example.com.csr -out ${wd}/httpbin.example.com.crt
+        # shellcheck disable=SC2086
+        kubectl create -n istio-system secret generic "${credential_name}" --from-file=key=${wd}/httpbin.example.com.key --from-file=cert=${wd}/httpbin.example.com.crt --cluster "${CLUSTER}"
+    done
+}
+
+timestamp=$(date +"%Y-%m-%d-%H-%M-%S")
+testns=httpbin-${timestamp}
+function deploy_httpbin() {
+    kubectl create ns "${testns}" --cluster "${CLUSTER}"
+    kubectl apply -n "${testns}" --cluster "${CLUSTER}" -f https://raw.githubusercontent.com/istio/istio/release-1.5/samples/httpbin/httpbin.yaml
+}
+
+function deploy_gateways() {
+    # shellcheck disable=SC2004
+    for ((id=1; id<=${NUM}; id++)); do
+        deploy_gateway $id "${testns}" "${CLUSTER}"
+        deploy_virtualservice $id "${testns}" "${CLUSTER}"
+    done
+}
+
+function deploy_clients() {
+    # shellcheck disable=SC2004
+    for ((id=1; id<=${NUM}; id++)); do
+        deploy_sleep $id "${testns}" "${CLUSTER}" "${wd}" "${secure_ingress_port}"
+    done
+}
+
+function check_access() {
+    # shellcheck disable=SC2004
+    for ((id=1; id<=${NUM}; id++)); do
+        host="httpbin-${id}.example.com"
+        # shellcheck disable=SC2153
+        url="https://httpbin-${id}.example.com:${secure_ingress_port}/status/418"
+        n=0
+        until [ "$n" -ge 10 ]
+        do
+          # shellcheck disable=SC2153
+          resp_code=$(curl -sS  -o /dev/null -w "%{http_code}\n" -HHost:"${host}" --resolve "${host}":"${secure_ingress_port}":"${ingress_host}" --cacert "${wd}"/example.com.crt "${url}")
+          # shellcheck disable=SC2086
+          if [ ${resp_code} = 418 ]; then
+            echo "${host}: OK"
+            break
+          elif [ "$n" = 10 ]; then
+            echo echo "${host} not ready"
```

Comment:
Can we do better than just indicating the error via stdout?

Reply:
If this call fails after 20 times, which is 10 seconds, then it requires investigation. Just terminate the test.